pkg/client/image_test.go

```diff
@@ -0,0 +1,323 @@
+package client_test
+
+// TODO: Add failure states to be tested
+
+import (
+	"context"
+
+	v12 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	"github.com/acorn-io/acorn/pkg/client"
+	"github.com/acorn-io/acorn/pkg/labels"
+	"github.com/acorn-io/acorn/pkg/mocks"
+	scheme2 "github.com/acorn-io/acorn/pkg/scheme"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	// may want to use envtest instead
+	"testing"
+
+	testcontrollerclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+type defaultClientImagesConfig struct {
+	images      v12.ImageList
+	projectName string
+	namespace   string
+}
+
+func createMockedDefaultClientImageLister(t *testing.T, config defaultClientImagesConfig) (client.DefaultClient, error) {
+	t.Helper()
+	ns := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: config.projectName,
+			Labels: map[string]string{
+				"test.acorn.io/namespace": "true",
+				labels.AcornProject:       "true",
+			},
+		},
+		Spec:   corev1.NamespaceSpec{},
+		Status: corev1.NamespaceStatus{},
+	}
+
+	imageListObj := config.images
+	testingScheme := scheme2.Scheme
+	err := scheme2.AddToScheme(testingScheme)
+	if err != nil {
+		return client.DefaultClient{}, err
+	}
```

Comment:
Not needed. This is done when you import the package.

Reply:
You're right. Thanks!!